### ISSUE TITLE
exit status 3221225781

### ISSUE THREAD

**Issue by erihaydee** [rana/ora#60]
Hi!, I am getting exit status 3221225781 just to load the library.

My oracle client is in the PATH, and I am running x64 versions of everything.  

Trying to run the application I get 
"The program can't start because OCI.dll is missing from your computer.  Try reinstalling the program to fix this problem"

CGO_CFLAGS=-IC:\instantclient_12_1\sdk\include
CGO_LDFLAGS=C:\instantclient_12_1\oci.dll
Path=.;C:\instantclient_12_1;C:\Windows\system32;C:\Windows\system32;C:\Windows;C
:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program F
iles (x86)\Microsoft Application Virtualization Client;c:\Program Files (x86)\Mi
crosoft SQL Server\100\Tools\Binn\;c:\Program Files\Microsoft SQL Server\100\Too
ls\Binn\;c:\Program Files\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files (x
86)\Microsoft ASP.NET\ASP.NET Web Pages\v1.0\;C:\Program Files\Microsoft SQL Ser
ver\110\Tools\Binn\;c:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\M
anagementStudio\;c:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\;c:\
Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\Mi
crosoft SDKs\TypeScript\;C:\Program Files (x86)\Windows Kits\8.1\Windows Perform
ance Toolkit\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program F
iles (x86)\Microsoft SDKs\TypeScript\1.0\;C:\Program Files (x86)\WinMerge;C:\Pro
gram Files\TortoiseGit\bin;%TFSPowerToolDir%;%BPADir%;C:\Program Files (x86)\Sky
pe\Phone\;C:\Program Files (x86)\nodejs\;C:\Program Files (x86)\QuickTime\QTSyst
em\;C:\Go\bin
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC

**Comment by tgulacsi** [comment]
The CGO_LDFLAGS is wrong, it has to be "-LC:\instantclient_12_1", maybe with "-loci".

**Comment by erihaydee** [comment]
thanks for coming back, 

I tried to compile again the library with the changes to the CGO_LDFLAGS and now is not compiling.

C:\Projects\Go\src>SET CGO_LDFLAGS
CGO_LDFLAGS=-LC:\instantclient_12_1\oci.dll

# gopkg.in/rana/ora.v3
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndBfile.cgo2.o:bndBfile.cgo2.c:(.text+0xc9): undefined reference to `OCIBindByPos2'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndBfile.cgo2.o:bndBfile.cgo2.c:(.text+0x150): undefined reference to `OCIDescriptorAlloc'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndBfile.cgo2.o:bndBfile.cgo2.c:(.text+0x1ad): undefined reference to `OCIDescriptorFree'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndBfile.cgo2.o:bndBfile.cgo2.c:(.text+0x24e): undefined reference to `OCILobFileSetName'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndBinSlice.cgo2.o:bndBinSlice.cgo2.c:(.text+0x5b): undefined reference to `OCIBindArrayOfStruct'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndFloat32.cgo2.o:bndFloat32.cgo2.c:(.text+0x45): undefined reference to `OCINumberFromReal'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndFloat32Ptr.cgo2.o:bndFloat32Ptr.cgo2.c:(.text+0x45): undefined reference to `OCINumberToReal'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndInt16.cgo2.o:bndInt16.cgo2.c:(.text+0x4d): undefined reference to `OCINumberFromInt'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndInt16Ptr.cgo2.o:bndInt16Ptr.cgo2.c:(.text+0x4d): undefined reference to `OCINumberToInt'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndIntervalDS.cgo2.o:bndIntervalDS.cgo2.c:(.text+0x7d): undefined reference to `OCIIntervalSetDaySecond'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndIntervalYM.cgo2.o:bndIntervalYM.cgo2.c:(.text+0x4d): undefined reference to `OCIIntervalSetYearMonth'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndLob.cgo2.o:bndLob.cgo2.c:(.text+0x8d): undefined reference to `OCILobCreateTemporary'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndLob.cgo2.o:bndLob.cgo2.c:(.text+0xf7): undefined reference to `OCILobFreeTemporary'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndLob.cgo2.o:bndLob.cgo2.c:(.text+0x1f6): undefined reference to `OCILobWrite2'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\bndTime.cgo2.o:bndTime.cgo2.c:(.text+0xd1): undefined reference to `OCIDateTimeConstruct'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defBfile.cgo2.o:defBfile.cgo2.c:(.text+0xa8): undefined reference to `OCIDefineByPos2'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defBfile.cgo2.o:defBfile.cgo2.c:(.text+0x148): undefined reference to `OCILobFileGetName'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defIntervalDS.cgo2.o:defIntervalDS.cgo2.c:(.text+0x80): undefined reference to `OCIIntervalGetDaySecond'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defIntervalYM.cgo2.o:defIntervalYM.cgo2.c:(.text+0x4d): undefined reference to `OCIIntervalGetYearMonth'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defLob.cgo2.o:defLob.cgo2.c:(.text+0x3b): undefined reference to `OCILobClose'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defLob.cgo2.o:defLob.cgo2.c:(.text+0xaf): undefined reference to `OCILobGetLength2'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defLob.cgo2.o:defLob.cgo2.c:(.text+0x127): undefined reference to `OCILobOpen'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defLob.cgo2.o:defLob.cgo2.c:(.text+0x226): undefined reference to `OCILobRead2'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defLob.cgo2.o:defLob.cgo2.c:(.text+0x2a6): undefined reference to `OCILobTrim2'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defTime.cgo2.o:defTime.cgo2.c:(.text+0x5d): undefined reference to `OCIDateTimeGetDate'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defTime.cgo2.o:defTime.cgo2.c:(.text+0xf8): undefined reference to `OCIDateTimeGetTime'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defTime.cgo2.o:defTime.cgo2.c:(.text+0x173): undefined reference to `OCIDateTimeGetTimeZoneName'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\defTime.cgo2.o:defTime.cgo2.c:(.text+0x1ee): undefined reference to `OCIDateTimeGetTimeZoneOffset'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\env.cgo2.o:env.cgo2.c:(.text+0x5a): undefined reference to `OCIAttrSet'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\env.cgo2.o:env.cgo2.c:(.text+0xf4): undefined reference to `OCIErrorGet'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\env.cgo2.o:env.cgo2.c:(.text+0x16f): undefined reference to `OCIHandleAlloc'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\env.cgo2.o:env.cgo2.c:(.text+0x1cc): undefined reference to `OCIHandleFree'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\env.cgo2.o:env.cgo2.c:(.text+0x245): undefined reference to `OCIServerAttach'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\ora.cgo2.o:ora.cgo2.c:(.text+0x88): undefined reference to `OCIEnvCreate'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\ora.cgo2.o:ora.cgo2.c:(.text+0x163): undefined reference to `OCIEnvNlsCreate'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\ora.cgo2.o:ora.cgo2.c:(.text+0x1c4): undefined reference to `OCINlsCharSetNameToId'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\rset.cgo2.o:rset.cgo2.c:(.text+0x5a): undefined reference to `OCIAttrGet'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\rset.cgo2.o:rset.cgo2.c:(.text+0xd2): undefined reference to `OCIParamGet'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\rset.cgo2.o:rset.cgo2.c:(.text+0x17d): undefined reference to `OCIStmtExecute'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\rset.cgo2.o:rset.cgo2.c:(.text+0x20b): undefined reference to `OCIStmtFetch2'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\ses.cgo2.o:ses.cgo2.c:(.text+0x30): undefined reference to `OCIBreak'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\ses.cgo2.o:ses.cgo2.c:(.text+0x98): undefined reference to `OCIPing'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\ses.cgo2.o:ses.cgo2.c:(.text+0x10c): undefined reference to `OCISessionEnd'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\ses.cgo2.o:ses.cgo2.c:(.text+0x1c1): undefined reference to `OCIStmtPrepare2'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\ses.cgo2.o:ses.cgo2.c:(.text+0x236): undefined reference to `OCITransStart'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\srv.cgo2.o:srv.cgo2.c:(.text+0x3a): undefined reference to `OCIServerDetach'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\srv.cgo2.o:srv.cgo2.c:(.text+0xb7): undefined reference to `OCIServerVersion'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\srv.cgo2.o:srv.cgo2.c:(.text+0x130): undefined reference to `OCISessionBegin'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\stmt.cgo2.o:stmt.cgo2.c:(.text+0x4b): undefined reference to `OCIStmtRelease'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\tx.cgo2.o:tx.cgo2.c:(.text+0x3a): undefined reference to `OCITransCommit'
C:\Users\erm01\AppData\Local\Temp\go-build105510710\gopkg.in\rana\ora.v3\_obj\tx.cgo2.o:tx.cgo2.c:(.text+0xa2): undefined reference to `OCITransRollback'
collect2.exe: error: ld returned 1 exit status

------

C:\Projects\Go\src>SET CGO_LDFLAGS
CGO_LDFLAGS=-LC:\instantclient_12_1\oci.dll -loci

# gopkg.in/rana/ora.v3
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -loci
collect2.exe: error: ld returned 1 exit status

I am using a Windows 7 64bit enviroment.

**Comment by tgulacsi** [comment]
PLEASE read my previous response, and follow it!

TL;DR; CGO_LDFLAGS does not need oci.dll in the -L flag!

**Comment by erihaydee** [comment]
Thanks a lot!, it works.
